lib/cmdline.c

```diff
@@ -1379,13 +1286,17 @@ bool rm_cmd_parse_args(int argc, char **argv, RmSession *session) {
         {"partial-hidden"           , 0    , EMPTY     , G_OPTION_ARG_CALLBACK  , FUNC(partial_hidden)           , _("Find hidden files in duplicate folders only")                          , NULL}     ,
         {"mtime-window"             , 'Z'  , 0         , G_OPTION_ARG_DOUBLE    , &cfg->mtime_window             , _("Consider duplicates only equal when mtime differs at max. T seconds")  , "T"}      ,
 
+        /* COW filesystem deduplication support */
+        {"dedupe"                   , 0    , 0         , G_OPTION_ARG_NONE      , &cfg->dedupe                   , _("Dedupe matching extents from source to dest (if filesystem supports)") , NULL}     ,
+        {"dedupe-readonly"          , 'r'  , 0         , G_OPTION_ARG_NONE      , &cfg->dedupe_readonly          , _("(--dedupe option) even dedupe read-only snapshots (needs root)")       , NULL}     ,
```

Comment:
Yeah but `--dedupe` is a subcommand which is mutually exclusive with the main rmlint.  The `-r` shortcut will clear `cfg->ignore_hidden` and also set `cfg->dedupe_readonly`.  This will work as expected for both `rmlint -r` and `rmlint --dedupe -r`.  It's a bit hacky but it works.
I did start playing around with `GOptionGroup`'s with the idea of formalising the Usage along the lines of:
```
Usage:
rmlint [options]           Search for duplicate files and other lint; see rmlint --help
rmlint dedupe [options]    Dedupe matching extents of two [or more] files; see rmlint --help-dedupe
rmlint gui                 Start rmlint gui app
etc
```
For now I think it works as-is but maybe I'll come back to this later